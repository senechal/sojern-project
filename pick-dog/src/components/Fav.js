import {  useSelector } from "react-redux";
import Galery from './Galery';

const Fav = () => {
    const { imgs } = useSelector(state => state.fav);

    return <Galery imgs={imgs} />
};

export default Fav;