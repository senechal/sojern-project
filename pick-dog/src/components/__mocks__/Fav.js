const Fav = () => {
    return <div>Fav Mock</div>
};

export default Fav;