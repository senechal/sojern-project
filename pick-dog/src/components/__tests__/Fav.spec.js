import { useSelector } from 'react-redux';
import { render } from '@testing-library/react';
import Fav from '../Fav';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

describe('Fav tests', () => {

  it('should, render when empty', () => {
    useSelector.mockReturnValueOnce({ imgs: []});
    const { container } = render(<Fav />);
    expect(container).toMatchSnapshot();
  });
  it('should, render when not empty', () => {
    useSelector.mockReturnValueOnce({ imgs: [{id: '1', url: 'http://mock_image/1.jpg', type: 'jpg'}]});
    const { container } = render(<Fav />);
    expect(container).toMatchSnapshot();
  });
});
