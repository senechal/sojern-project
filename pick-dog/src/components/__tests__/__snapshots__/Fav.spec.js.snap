// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fav tests should, render when empty 1`] = `
<div>
  <div
    class="sc-bdfBwQ bWYwLj"
  />
</div>
`;

exports[`Fav tests should, render when not empty 1`] = `
<div>
  <div
    class="sc-bdfBwQ bWYwLj"
  >
    <img
      alt="http://mock_image/1.jpg"
      class="sc-gsTCUz ivfovS"
      data-id="1"
      data-testid="galery-image"
      loading="lazy"
      src="http://mock_image/1.jpg"
    />
  </div>
</div>
`;
